sled-agent/src/bootstrap/agent.rs

```diff
@@ -89,74 +125,84 @@ impl Agent {
 
     /// Communicates with peers, sharing secrets, until the rack has been
     /// sufficiently unlocked.
-    ///
-    /// - This method retries until [`UNLOCK_THRESHOLD`] other agents are
-    /// online, and have successfully responded to "share requests".
-    async fn establish_sled_quorum(&self) -> Result<(), BootstrapError> {
-        retry_notify(
+    async fn establish_sled_quorum(
+        &self,
+    ) -> Result<RackSecret, BootstrapError> {
+        let rack_secret = retry_notify(
             internal_service_policy(),
             || async {
                 let other_agents = self.peer_monitor.addrs().await;
-                info!(&self.log, "Bootstrap: Communicating with peers: {:?}", other_agents);
+                info!(
+                    &self.log,
+                    "Bootstrap: Communicating with peers: {:?}", other_agents
+                );
+
+                let share = || self.share.as_ref().unwrap();
```

Comment:
Why hide this behind a lambda? Can't we just do:

```suggestion
                let share = self.share.as_ref().unwrap();
```

Reply:
Good catch. I have no idea why I did this. Things that make you go hmm. Fixed.